src/query-builder/QueryBuilder.ts

```diff
@@ -730,10 +730,10 @@ export abstract class QueryBuilder<Entity> {
 
         } else {
             const [primaryColumn] = metadata.primaryColumns;
-            const areAllNumbers = normalized.every((id: any) => typeof id === "number");
+            const normalizedValues = normalized.map(ent => ent[Object.keys(ent)[0]]);
+            const areAllNumbers = normalizedValues.every((id: any) => typeof id === "number");
             if (areAllNumbers) {
-                return `${alias + this.escape(primaryColumn.databaseName)} IN (${normalized.join(", ")})`;
-
+                return `${alias + this.escape(primaryColumn.databaseName)} IN (${normalizedValues.join(", ")})`;
```

Comment:
This was a dead code. Still not sure if we should put number parameters inside query string. Parametrized queries can use already compiled query plans(performance).